### PR TITLE
Feature/order dishes

### PR DESCRIPTION
**Pull request by AlexanderWidmann**
feat(create-order): user can now order a list of dishes at a selected time and canteen
feat(display-order): user can display all orders and/or one specific order

### CODE REVIEW

File: src/main/java/edu/aau/groupc/canteenbackend/mgmt/Canteen.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    @Id
+    @Column(name = "CANTEEN_ID")
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Integer id;
+    @NotNull
+    private String name;
+    @NotNull
+    private String address;
+    private int numTables;
+    @OneToMany(mappedBy = "canteen")
```
Comment (elsantner) [inline src/main/java/edu/aau/groupc/canteenbackend/mgmt/Canteen.java]: Is this not configured to be fetched lazily (i.e. no `fetch = FetchType.LAZY`) on purpose?

File: src/main/java/edu/aau/groupc/canteenbackend/orders/dto/CreateOrderDTO.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    boolean reserveTable = false;
+
+    @NotNull(message = "CanteenId is required")
+    private Integer canteenId;
+
+    @NotNull(message = "Dish-List is required")
+    @Size(min = 1, message = "Dish-List must not be empty")
+    private List<@Valid DishForOrderCreationDTO> dishes;
+    @NotNull
+    @PickupdateConstraint
```
Comment (elsantner) [inline src/main/java/edu/aau/groupc/canteenbackend/orders/dto/CreateOrderDTO.java:26]: Very nice solution! I like it a lot!

File: src/main/java/edu/aau/groupc/canteenbackend/user/User.java [hunk 3]
```diff
@@ -22,6 +26,9 @@
     private String username;
     private String password;
     private Type type;
+    @OneToMany(mappedBy = "user")
```
Comment (elsantner) [inline src/main/java/edu/aau/groupc/canteenbackend/user/User.java]: Is `fetch = LAZY` not configured on purpose?

**Review by elsantner — COMMENTED** [review]
Overall very nicely implemented imo!
I don't see any real issues with it. I've left a comment regarding lazy fetching which might have some effects on performance.

Also, as far as I see all remaining `TODO` comments rely on other features, right? So these would have to be implemented in a separate PR once the required functionality is implemented.

(I'm just waiting for a reply regarding the lazy fetching before I approve)

File: src/main/java/edu/aau/groupc/canteenbackend/mgmt/Canteen.java (hunk 1 above)
Comment (elsantner) [reply src/main/java/edu/aau/groupc/canteenbackend/mgmt/Canteen.java]: Ok, nevermind! Turns out the default fetch type for `OneToMany` is already `LAZY` ([Source](https://stackoverflow.com/a/26601720)).

**Review by elsantner — APPROVED** [review]

File: src/main/java/edu/aau/groupc/canteenbackend/user/User.java (hunk 3 above)
Comment (AlexanderWidmann) [reply src/main/java/edu/aau/groupc/canteenbackend/user/User.java]: It is the default, but i added it to keep the code consistent

**Pull request merged**